src/tlo/methods/hiv.py

```diff
@@ -1504,7 +1504,7 @@ def per_capita_testing_rate(self):
         df = self.sim.population.props
 
         # get number of tests performed in last time period
-        if self.sim.date.year == 2011:
+        if self.sim.date.year == (self.sim.start_date.year + 1):
```

Comment:
per_capita_testing_rate is called by HivLoggingEvent, which first occurs 01/01/2011, i.e. one year after sim start. The function uses the previous years number of tests performed to calculate the number of tests performed in this time period. So for the first year, the value of previous_test_numbers must equal 0. 

I think a workaround could be:

# Check if it's the first year of the simulation and no results yet stored in self.stored_test_numbers

if not self.stored_test_numbers:
    # If it's the first year, set previous_test_numbers to 0
    previous_test_numbers = 0
else:
    # For subsequent years, retrieve the last stored number
    previous_test_numbers = self.stored_test_numbers[-1]

# Calculate number of tests now performed - cumulative, include those who have died
number_tests_new = df.hv_number_tests.sum()

# Store the number of tests performed in this year for future reference
self.stored_test_numbers.append(number_tests_new)

# Number of tests performed in the last time period
number_tests_in_last_period = number_tests_new - previous_test_numbers

Please note - I haven't tested this!

Reply:
ok no probs - let me try that